File: Tema 2'/EditieLimitata.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tema_2_
{
    class EditieLimitata : ProdusVechi
    {
        public string DataFinalOferta { get; set; }
        public int NrProdusePeClient { get; set; }
        public void Afisare(char C)
        {
            Console.WriteLine($"Produsul are caracterul de identificare {C} (Polimorfism static)\n");
            D.AfisareDetalii();
            Console.WriteLine();
        }
        public override void Afiseaza()
        {
            Console.WriteLine("Produs din clasa <<EditieLimitata>> (Polimorfism dinamic)\n");
            D.AfisareDetalii();
            Console.WriteLine();
        }
        public override void Afis()
        {
            Console.WriteLine("Produs din clasa <<EditieLimitata>> (Interfata)\n");
        }
    }
}

File: C++/EditieLimitata.cpp
#include "EditieLimitata.h"
#include <iostream>
using namespace std;
void EditieLimitata::afisare(char c) const
{
	cout << "Produsul are caracterul de identificare: " << c <<"(Polimorfism static)" << endl;
	d.afisareDetalii();
	cout << endl;
}
void EditieLimitata::afiseaza() const
{
	cout << "Produs din clasa <<EditieLimitata>>(Polimorfism dinamic)" << endl;
	d.afisareDetalii();
	cout << endl;
}
void EditieLimitata::afis()const
{
	cout << "Produs de clasa EditieLimitata (Interfata)" << endl;
	cout << endl;
}